riff-raff/app/schedule/DeployJob.scala

```diff
@@ -56,8 +52,12 @@ object DeployJob extends Logging with LogAndSquashBehaviour {
         } else {
           Left(Error(s"Scheduled deployments disabled. Would have deployed $params"))
         }
+      case RunState.Failed =>
+        Left(defaultError(RunState.Failed).copy(scheduledDeployError = Some(SkippedDueToPreviousFailure(lastDeploy))))
```

Comment:
You could avoid the copy by adding another parameter with a default value to `defaultError` - perhaps renamed to `stateError` or some such.
This might look different with a common error hierarchy.